### PR TITLE
Created Shape class

### LINKED ISSUE #4: Set up Shape abstract class
As a developer, I want shapes to inherit from a single class, so that new functionality that affects all shapes is easy to implement

Tasks:
 - Shape class with abstract methods
 - Shape has pure virtual functions draw, move, perimeter & area
 - Overload the equality and less than relational operators in base class Shape
 - Disable copy operations
 - Each shape stores its position on the canvas
 - Shape has a virtual toString() method so that the reporter can enumerate all shape properties
 - Shape tracks a unique ID number for each shape

Tests:
 - A derived class of Shape can be instantiated
 - A derived class can be placed on the canvas and rendered
 - Overlapping shapes do not z-fight

Priority: High

### PR DESCRIPTION
**Pull request by ChrisRzech**
Tell me if more stuff is needed. Like keeping track of all the shapes might be a good one, but I'm not sure if you guys want that.

Closes #4

### CODE REVIEW

**Comment by dacruz21** [comment]
Also I think we're using <cstdint> integers now -- Joji's idiot compiler doesn't like `uint`

File: shapes/abstshape.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#include "abstshape.hpp"
```
[the remark this answers is not in the document]
Comment (ChrisRzech) [reply shapes/abstshape.cpp]: what you talking about willis?

File: Jojibrush.pro [hunk 2]
```diff
@@ -25,10 +25,12 @@ DEFINES += QT_DEPRECATED_WARNINGS
 
 SOURCES += \
         main.cpp \
-        mainwindow.cpp
+        mainwindow.cpp \
+    shapes/shape.cpp
```
Comment (dacruz21) [inline Jojibrush.pro]: Can you match indentation here please?

File: shapes/shape.hpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    /* Destructor */
+    virtual ~Shape();
+
+    /* Position */
+    QPoint getPosition() const;
+    void   setPosition(QPoint);
+
+    /* Render */
+    virtual void draw() const = 0;
```
Comment (dacruz21) [inline shapes/shape.hpp]: This definitely needs to have a QWidget parameter so the Shape knows what surface to draw to

File: shapes/shape.hpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+#include <QString>
+#include <string>
+
+namespace jbrush
+{
+class Shape
+{
+public:
+    /* Constructors */
+    Shape(QPoint);
```
Comment (dacruz21) [inline shapes/shape.hpp]: We need to disable copy instructions here and at the operators below

Comment (ChrisRzech) [reply shapes/shape.hpp]: so you're saying to put the deleted members in their respective sections?

```cpp
    /* Constructors */
    Shape(QPoint);
    Shape(const Shape&) = delete;
```

File: shapes/shape.hpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    /* Destructor */
+    virtual ~Shape();
+
+    /* Position */
+    QPoint getPosition() const;
```
Comment (dacruz21) [inline shapes/shape.hpp]: need a getter (but not setter) for ID

**Pull request merged**